atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -120,71 +131,110 @@ public synchronized void storeUpperLimit(final long limit) {
         clientPool.runWithRetry(new FunctionCheckedException<Client, Void, RuntimeException>() {
             @Override
             public Void apply(Client client) {
-                cas(client, currentLimit, limit);
+                cas(client, makeColumnWithNewFormat(getId(), currentLimit), currentLimit, limit);
                 return null;
             }
         });
     }
 
-    private void cas(Client client, Long oldVal, long newVal) {
-        final CASResult result;
+    private void cas(Client client, Column oldColumn, Long oldVal, long newVal) {
+        CASResult result = updateTimestampInDb(client, oldColumn, oldVal, newVal);
+        if (result.isSuccess()) {
+            DebugLogger.logger.info("[CAS] Setting cached limit to {}.", newVal);
+            currentLimit = newVal;
+        } else {
+            if (result.getCurrent_values().isEmpty()) {
+                log.error("[CAS] Error trying to set timestamp limit to {}: there is no limit stored in DB.", newVal);
+                throw new MultipleRunningTimestampServiceError("Error trying to set timestamp limit: there is no "
```

Comment:
It happens when you try to change a bound and the table is empty -- which is allowed for reads. We throw a MRTSE because that was the behaviour before and the interface API says so -- I'd be happy to change it to something more reasonable if you're fine with that change.